pkg/controllers/gpu/controller.go

```diff
@@ -170,6 +170,64 @@ func (c *controller) NewDaemonSet(rawObj runtime.Object) *apps.DaemonSet {
 			},
 		},
 	}
+	// add the optional InitImage
+	if devicePlugin.Spec.InitImage != "" {
+		daemonSet.Spec.Template.Spec.InitContainers = createInitContainer(devicePlugin.Spec.InitImage)
+		appendVolumeIfMissing(&daemonSet.Spec.Template.Spec.Volumes, "nfd-source-hooks")
```

Comment:
Sine these two functions are always used together I'd unite them into one e.g. `setInitContainer(&daemonSet.Spec.Template.Spec, devicePlugin.Spec.InitImage)` . And there's no need to parameterize it with a volume name.